hyperlane-mailbox/tests/harness.rs

```diff
@@ -180,3 +212,121 @@ async fn test_latest_checkpoint() {
     // The index is 0-indexed
     assert_eq!(index, 0u32);
 }
+
+#[tokio::test]
+async fn test_process_id() {
+    let (mailbox, ism_id, recipient_id, _) = get_contract_instance().await;
+
+    let metadata = vec![5u8; 100];
+
+    let agent_message = test_message(&mailbox, recipient_id.clone(), false);
+    let agent_message_id = agent_message.id();
+
+    let contract_inputs = vec![ism_id.clone(), recipient_id];
+
+    let process_call = mailbox
+        .methods()
+        .process(
+            metadata.clone(),
+            agent_message.clone().into(),
+        )
+        .set_contracts(&contract_inputs)
+        .tx_params(TxParameters::new(None, Some(1_200_000), None))
+        .call()
+        .await
+        .unwrap();
+        
+    let message_id = &process_call.get_logs_with_type::<Bits256>().unwrap()[0];
+    
```

Comment:
Wondering if we should check some kind of condition resulting from the recipient's `handle` function being called as an extra sanity check